purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -127,15 +127,35 @@ def _compute_qty_refunded(self):
         'move_ids.returned_move_ids.state',
     )
     def _compute_qty_returned(self):
+        """Made through read_group for not impacting in performance."""
+        ProductUom = self.env['product.uom']
+        StockMove = self.env['stock.move']
+        groups = StockMove.read_group(
+            [('origin_returned_move_id.purchase_line_id', 'in', self.ids),
+             ('state', '=', 'done'),
+             ('location_id.usage', '!=', 'supplier')],
+            ['origin_returned_move_id', 'product_uom_qty', 'product_uom'],
+            ['origin_returned_move_id', 'product_uom'], lazy=False,
+        )
+        p = self._prefetch
+        # load all moves records at once on first access
+        origin_move_ids = set(g['origin_returned_move_id'][0] for g in groups)
+        StockMove.browse(origin_move_ids).with_prefetch(p)
+        # load all UoM records at once on first access
+        uom_ids = set(g['product_uom'][0] for g in groups)
+        ProductUom.browse(list(uom_ids)).with_prefetch(p)
+        line_qtys = {}
+        for g in groups:
+            uom = ProductUom.browse(g["product_uom"][0]).with_prefetch(p)
+            move = StockMove.browse(
+                g['origin_returned_move_id'][0]
+            ).with_prefetch(p)
+            line_qtys.setdefault(move.purchase_line_id.id, 0)
+            line_qtys[move.purchase_line_id.id] += uom._compute_quantity(
```

Comment:
Add condition `uom == move.purchase_line_id.product_uom` to set g['product_uom_qty'] directly instead of call _compute_quantity function

Reply:
The same is evaluated on the `_compute_quantity` method, so no gain adding that lines

Reply:
I don't see:
https://github.com/odoo/odoo/blob/3112cd2dbe7459739f00eb775d5510f0aefcb3c2/addons/product/models/product_uom.py#L91-L106

Reply:
Don't compute_quantity if it is the same uom